### PR TITLE
projector: visualize_embeddings API change

### PR DESCRIPTION
**Pull request by stephanwlee**
Previously, it used to take an FileWriter as an input so it can infer
logdir to write embedding to. Because summary v2 API deprecated an
ability to get the logdir, we now instead take the logdir as string.

To work with the older code, we make sure we can also take the v1
summary writer.

Addresses #2471, aside from supporting the new checkpoint naming

### CODE REVIEW

File: tensorboard/plugins/projector/projector_api_test.py [hunk 1]
```diff
@@ -21,37 +21,66 @@
...
+    logdir = self.get_temp_dir()
+    self.addCleanup(shutil.rmtree, logdir)
+
+    config = create_dummy_config()
+    projector.visualize_embeddings(logdir, config)
+
+    # Read the configurations from disk and make sure it matches the original.
+    with tf.io.gfile.GFile(os.path.join(logdir, 'projector_config.pbtxt')) as f:
+      config2 = projector.ProjectorConfig()
+      # config2.ParseFromString(f.read())
```
Comment (wchargin) [inline tensorboard/plugins/projector/projector_api_test.py]: Delete?

File: tensorboard/plugins/projector/projector_api_test.py [hunk 2]
```diff
@@ -21,37 +21,66 @@
...
-  def testVisualizeEmbeddings(self):
-    # Create a dummy configuration.
-    config = projector.ProjectorConfig()
-    config.model_checkpoint_path = 'test'
-    emb1 = config.embeddings.add()
-    emb1.tensor_name = 'tensor1'
-    emb1.metadata_path = 'metadata1'
+  def test_visualize_embeddings_with_logdir(self):
+    logdir = self.get_temp_dir()
+    self.addCleanup(shutil.rmtree, logdir)
```
Comment (wchargin) [inline tensorboard/plugins/projector/projector_api_test.py]: No need: `get_temp_dir()` is cleaned up automatically, and every `rm -r`
that we can remove from the codebase makes me sleep a little easier. :-)

File: tensorboard/plugins/projector/projector_api_test.py [hunk 3]
```diff
@@ -21,37 +21,66 @@
...
 from google.protobuf import text_format
 
 from tensorboard.plugins import projector
 from tensorboard.util import test_util
 
-tf.compat.v1.disable_v2_behavior()
-
+def create_dummy_config():
+  return projector.ProjectorConfig(
+      model_checkpoint_path = 'test',
```
Comment (wchargin) [inline tensorboard/plugins/projector/projector_api_test.py]: nit: No spaces around `kwarg=value`s in Python.

File: tensorboard/plugins/projector/projector_api_test.py (hunk 2 above)
Comment (stephanwlee) [reply tensorboard/plugins/projector/projector_api_test.py]: Did not know! Thanks!

File: tensorboard/plugins/projector/projector_api_test.py (hunk 3 above)
Comment (stephanwlee) [reply tensorboard/plugins/projector/projector_api_test.py]: I got lazy and expected formatter to kick in :( I am spoiled!

**Pull request merged**